visualization/motion_planning_tasks/src/remote_task_model.cpp

```diff
@@ -180,13 +184,13 @@ QModelIndex RemoteTaskModel::index(const Node* n) const {
 	return QModelIndex();
 }
 
-RemoteTaskModel::RemoteTaskModel(ros::NodeHandle& nh, const std::string& service_name,
-                                 const planning_scene::PlanningSceneConstPtr& scene,
-                                 rviz::DisplayContext* display_context, QObject* parent)
+RemoteTaskModel::RemoteTaskModel(const std::string& service_name, const planning_scene::PlanningSceneConstPtr& scene,
+                                 rviz_common::DisplayContext* display_context, QObject* parent)
   : BaseTaskModel(scene, display_context, parent), root_(new Node(nullptr)) {
 	id_to_stage_[0] = root_;  // root node has ID 0
 	// service to request solutions
-	get_solution_client_ = nh.serviceClient<moveit_task_constructor_msgs::GetSolution>(service_name);
+	node_ = rclcpp::Node::make_shared("get_solution_node", "/moveit_task_constructor/remote_task_model");
```

Comment:
Shouldn't this use a node passed with the constructor? Also, the node probably needs an executor

Reply:
See my note [here](https://github.com/JafarAbdi/moveit_task_constructor/blob/ros2/visualization/motion_planning_tasks/src/remote_task_model.h#L59) on why I created an internal node

Reply:
There's no need for an executor when can spin it as needed